Bottom: a65c20292323863b962926c64bc57f6813a7ee07
Top:    ed2d60eef1dbef8da68edcc3b32e55baa26a9c9c
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-02-15 19:47:54 +0000

Tmp changes


---

diff --git a/src/polygon1.c b/src/polygon1.c
index 866c605..544fc6d 100644
--- a/src/polygon1.c
+++ b/src/polygon1.c
@@ -1707,12 +1707,14 @@ Collect (char poly, jmp_buf * e, PLINE * a, POLYAREA ** contours, PLINE ** holes
 //  cur = (&a->head)->next->next->next->next->next; /* Breaks circ_seg_test9.pcb */
   do
     {
+#if 0
       // The following may be a nice speedup, but not sure if it is correct.
       // In particular, consider the case when we collect a 'B' polygon contour.
       // Could some of that countour may already have been collected, and there
       // still be a piece we are interested in after? (Can we reach it though??)
       if (cur->Flags.mark != 0)
         break;
+#endif
 
       if (j_rule (poly, cur, &dir) && cur->Flags.mark == 0)
 	Collect1 (e, cur, dir, contours, holes, j_rule);
